pkg/policy/condition/stringequalsfunc.go

```diff
@@ -50,7 +51,26 @@ func (f stringEqualsFunc) evaluate(values map[string][]string) bool {
 		requestValue = values[f.k.Name()]
 	}
 
-	return !f.values.Intersection(set.CreateStringSet(requestValue...)).IsEmpty()
+	fvalues := set.NewStringSet()
```

Comment:
You would need to use https://godoc.org/github.com/minio/minio-go/pkg/set#StringSet.ApplyFunc.  If you would need to copy `StringSet` https://godoc.org/github.com/minio/minio-go/pkg/set#CopyStringSet is better (however I don't see a need to have copy)

Reply:
Well because I don't want to mutate original copy of map, instead want to mutate outside. Because values are dynamic and we don't want a orginal copy to be modified.

Reply:
This is what https://godoc.org/github.com/minio/minio-go/pkg/set#StringSet.ApplyFunc (and many methods) exactly doing.

Reply:
Well I have no idea to what extent Set is feature rich, it's just lack of understanding. Will use that..